bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java

```diff
@@ -51,13 +51,25 @@ public static BError createError(BString message) {
         return new ErrorValue(message);
     }
 
+    /**
+     * Create an error with given reason and details.
+     *
+     * @param message  error message
+     * @param details error details
+     * @return new error
+     */
+    public static BError createError(BString message, BMap<BString, Object> details) {
+        return new ErrorValue(message, details);
+    }
+
     /**
      * Create an error with given message and details.
      *
      * @param message error message
      * @param details error details
      * @return new error.
      */
+    @Deprecated
```

Comment:
Shall we add to use reateError(BString message, BMap<BString, Object> details) as a note in @deprecated annotation. Check runtime apis

Reply:
Shall we call createError(message, details) this method

Reply:
Added deprecation notes.